### PR TITLE
[DR-1592] Create dockerfile to setup cdn image

### PR DESCRIPTION
**Pull request by gpgpublickey**
## Background
  
Setup cdn images, install node, create npm package.json, download cdn uploader and install ftp dependency
  
## Changes done
  
Dockerfile created
Created CI travis file
Readme updated

## Pending to be done
  
Test if CI is working
  
## Notes
  
N/A

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM alpine:latest AS doppler-relay-akamai-publish
+WORKDIR /cdn
+RUN mkdir ./build
+RUN apk update
+RUN apk upgrade
+RUN apk add nodejs
+RUN wget https://raw.githubusercontent.com/MakingSense/doppler-docker-akamai-publish/master/cdn-uploader.js
```
Comment (andresmoschini) [inline Dockerfile]: Why do you prefer it in place of simply copy the script inside the image?

**Comment by andresmoschini** [comment]
Could you update README file in this PR indicating how to take advantage of this image?

File: Dockerfile (hunk 1 above)
Comment (joaquinicolas) [reply Dockerfile]: For perfomance reasons and good practice, is a better alternative make something like that: `RUN apk update && apk upgrade \\
apk add nodejs
`
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#sort-multi-line-arguments

Comment (andresmoschini) [reply Dockerfile]: Nice comment @joaquinicolas, maybe you could open a new thread for it. I think that the _wget_ vs _copy_ is a really important point and I do not like to miss it in this thread.

**Comment by gpgpublickey** [comment]
Of course

File: Dockerfile (hunk 1 above)
Comment (gpgpublickey) [reply Dockerfile]: I use wget to not depend of file routes. That we don't have defined yet. I suppose that a modular approach is better to test our current development.
We can change it later.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+language: bash
+services: docker
+
+install:
+  - git clone https://github.com/docker-library/official-images.git ~/official-images
+
+script:
+  - docker build -t "dopplerrelay/doppler-relay-akamai-publish" .
```
Comment (andresmoschini) [inline .travis.yml:8]: Could you also add a link in Readme file to CI and Docker Hub?

Comment (gpgpublickey) [reply .travis.yml:8]: Ready!

File: README.md [hunk 3]
```diff
@@ -1,2 +1,69 @@
...
+
+```js
+const config = {
+	host: process.env.AKAMAI_CDN_HOSTNAME,
+	user: process.env.AKAMAI_CDN_USERNAME,
+	password: process.env.AKAMAI_CDN_PASSWORD,
+	cpCode: process.env.AKAMAI_CDN_CPCODE,
+    port: 21,
+```
+}
```
Comment (andresmoschini) [inline README.md]: Maybe it is better to have an example of _.yml_ and not about where is being used.

**Comment by gpgpublickey** [comment]
Readme was recently updated @andresmoschini @mvillaseco

**Comment by andresmoschini** [comment]
Nice readme @cristiangiagante! It is really clear now.

**Comment by gpgpublickey** [comment]
Thanks to you @andresmoschini !

**Pull request merged**